lib/ansible/plugins/test/files.py

```diff
@@ -19,15 +19,20 @@
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
-from os.path import isdir, isfile, isabs, exists, lexists, islink, samefile, ismount
-from ansible import errors
+from os.path import isdir, isfile, isabs, exists, lexists, islink, samefile, ismount, expanduser, expandvars
 
 
 class TestModule(object):
     ''' Ansible file jinja2 tests '''
 
     def tests(self):
-        return {
+        def w(func):
```

Comment:
I would prefer that we stay away from nested functions like this if possible, moving into a scope where it could be tested independent of the `tests` method.

We should probably give it a better name too.  In `filters.core` we utilize a `unicode_wrap` function, so maybe we could do similarly here.

We also use `partial` there instead of creating our own closure, but either way is fine with me.

I'd also write tests to verify this functionality.

Reply:
also, this should not be by default as it changes current operations and might 'suprise' users that expect it to fail when the path is 'polluted'.

Reply:
@sivel @bcoca Good points, thanks. Fixed.